homeassistant/components/insteon_plm.py

```diff
@@ -26,95 +26,142 @@
+                                     'state_key': state_key}],
+                        hass_config=config))
 
     _LOGGER.info("Looking for PLM on %s", port)
-    plm = yield from insteonplm.Connection.create(device=port, loop=hass.loop)
+    conn = yield from insteonplm.Connection.create(
+        device=port,
+        loop=hass.loop,
+        workdir=hass.config.config_dir)
+
+    plm = conn.protocol
 
-    for device in overrides:
+    for device_override in overrides:
         #
         # Override the device default capabilities for a specific address
         #
-        if isinstance(device['platform'], list):
-            plm.protocol.devices.add_override(
-                device['address'], 'capabilities', device['platform'])
-        else:
-            plm.protocol.devices.add_override(
-                device['address'], 'capabilities', [device['platform']])
+        address = device_override.get('address', None)
+        if address is not None:
+            if device_override.get('cat', False):
+                plm.devices.add_override(address,
+                                         'cat',
+                                         device_override['cat'])
+            if device_override.get('subcat', False):
+                plm.devices.add_override(address,
+                                         'subcat',
+                                         device_override['subcat'])
+            if device_override.get('firmware', False):
+                plm.devices.add_override(address,
+                                         'product_key',
+                                         device_override['firmware'])
+            if device_override.get('product_key', False):
+                plm.devices.add_override(address,
+                                         'product_key',
+                                         device_override['product_key'])
 
     hass.data['insteon_plm'] = plm
 
-    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, plm.close)
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, conn.close)
 
-    plm.protocol.devices.add_device_callback(async_plm_new_device, {})
+    plm.devices.add_device_callback(async_plm_new_device)
 
     return True
 
 
-def common_attributes(entity):
+def common_attributes(entity, state):
     """Return the device state attributes."""
-    attributes = {}
-    attributekeys = {
-        'address': 'INSTEON Address',
-        'description': 'Description',
-        'model': 'Model',
-        'cat': 'Category',
-        'subcat': 'Subcategory',
-        'firmware': 'Firmware',
-        'product_key': 'Product Key'
+    attributes = {
+        'INSTEON Address': entity.address.human,
+        'Description': entity.description,
+        'Model': entity.model,
+        'Category': '{:02x}'.format(entity.cat),
+        'Subcategory': '{:02x}'.format(entity.subcat),
+        'Product Key / Firmware': '{:02x}'.format(entity.product_key),
```

Comment:
Firmware version and similar info is not allowed to be part of state attributes. The state machine should only hold attributes that describe the state of the entities.

Reply:
I made this change. Just so you know, this is meaningfully the same as it was in the original code just using the refactored approach to the insteonplm module. I see a lot of other platforms using the state machine the same way we were. I do think there is value to the user to be able to click on the entity and see the manufacturer, model number, etc. As more and more users start using HA we can expect they become less and less technical. Having that easily accessible I believe makes the platform more user friendly at a very low cost. (My two cents for what it is worth but will respect your direction.)